### PR TITLE
Switch improvements

### PR DESCRIPTION
**Pull request by PreslavKozovski**

### CODE REVIEW

File: packages/bootstrap/scss/_variables.scss [hunk 1]
```diff
@@ -986,6 +986,33 @@ $scrollview-navigation-default-opacity: .7 !default;
...
+$switch-handle-border-radius: $switch-size !default;
+$switch-handle-border-width: 1px !default;
+
+$swtich-on-handle-left: null !default;
+$swtich-off-handle-left: null !default;
+
+$switch-label-display: inline !default;
+$switch-label-spacing: 8px !default;
+
+$switch-occupied-space: add-three($switch-container-border-width, $switch-container-padding-x, $switch-handle-size);
```
Comment (PreslavKozovski) [inline packages/bootstrap/scss/_variables.scss]: @elena-gancheva @joneff 
The layout is pretty much finalized. Could you check these three variables, and let me know if you see a better way to these calculations?

**Comment by PreslavKozovski** [comment]
Merge this PR together with https://github.com/telerik/kendo-angular-inputs/issues/294 and https://github.com/telerik/kendo-react-private/issues/474

**Comment by jivanova** [comment]
- [ ] Implement switch labels positioning and dimension calculations. Switch labels will be moved outside of the k-switch-handle.

File: packages/default/scss/switch/_layout.scss [hunk 2]
```diff
@@ -1,99 +1,131 @@
...
+        @include border-radius( $switch-handle-border-radius );
         width: $switch-handle-size;
         height: $switch-handle-size;
-        display: inline-block;
-        margin: 0 ($switch-padding * 3) 0 $switch-padding;
-        border-width: $switch-border-width;
+        border-width: $switch-handle-border-width;
         border-style: solid;
+        display: inline-block;
         vertical-align: middle;
```
Comment (elena-gancheva) [inline packages/default/scss/switch/_layout.scss]: not needed

File: packages/default/scss/switch/_theme.scss [hunk 3]
```diff
@@ -47,73 +15,124 @@ $switch-active-shadow: 0 0 0 2px rgba(0, 0, 0, .06) !default;
...
         &.k-state-focused {
-            outline: none;
-            box-shadow: $switch-focus-shadow;
+            @include box-shadow( $switch-off-focused-shadow );
         }
 
-        &:active {
-            box-shadow: $switch-active-shadow;
+        .k-switch-label-on {
+            color: transparent;
```
Comment (elena-gancheva) [inline packages/default/scss/switch/_theme.scss]: do we need to assign a variable here for customization?

File: packages/default/scss/switch/_layout.scss [hunk 4]
```diff
@@ -1,99 +1,131 @@
...
     .k-switch {
+        @include border-radius( $switch-border-radius );
         cursor: pointer;
+        width: $switch-size;
+        background: none;
         border: 0;
-        display: inline-block;
-        font-size: $font-size-xs;
+        outline: 0;
+        box-shadow: none;
```
Comment (elena-gancheva) [inline packages/default/scss/switch/_layout.scss]: Shouldn't this be in `_theme.scss`?

File: packages/default/scss/switch/_layout.scss [hunk 5]
```diff
@@ -1,99 +1,131 @@
...
-        display: block;
-        text-align: center;
+        display: $switch-label-display;
+        width: $switch-label-width;
         position: absolute;
-        text-transform: uppercase;
+        top: 50%;
+        transform: translateY(-50%);
+        text-transform: $switch-label-text-transform;
         text-shadow: none;
```
Comment (elena-gancheva) [inline packages/default/scss/switch/_layout.scss]: `_theme.scss`

File: packages/default/scss/switch/_layout.scss (hunk 2 above)
Comment (PreslavKozovski) [reply packages/default/scss/switch/_layout.scss]: When removed, the handle is no longer aligned correctly.

File: packages/default/scss/switch/_theme.scss (hunk 3 above)
Comment (PreslavKozovski) [reply packages/default/scss/switch/_theme.scss]: No, we don't, this is the color of the label in the opposite state. For example, when the state is "OFF" we always want that the "ON" label is not present(transparent);

**Comment by tsvetomir** [comment]
If I understand correctly, these changes will render the theme unusable with older versions of the components. It should be marked as a **breaking change** and released as a major version.

**Pull request merged**